File: ChoiceTheory/R/EVT.R
# Options A and B (outcomes and probabilities are shown in vector)
x_A <-c(200, 22, 20) 
p_A <-c(0.9, 0.05, 0.05)
x_B <-c(200, 190, 20)
p_B <-c(0.85, 0.05, 0.1)

# Expected Value Theory (EVT)
# Function
EVT <- function (x,p){
	sum(x*p)
}

# Calculation
EVT_A <- EVT(x_A,p_A)
EVT_B <- EVT(x_B,p_B)

# Decision rule
if (EVT_A > EVT_B){
	print('Option A is chosen')
}else{
	print('Option B is chosen')
	}

File: ChoiceTheory/Matlab/EVT.m
%% Options A and B (outcomes and probabilities are shown in vector)
x_A=[200 22 20];
p_A=[.9 .05 .05];
x_B=[200 190 20];
p_B=[.85 .05 .1];

%% Expected Value Theory
% Calculation
EV_A = EVT(x_A, p_A);
EV_B = EVT(x_B, p_B);

% Decision rule
if EV_A > EV_B
    disp('EVT: Option A')
else
    disp('EVT: Option B')
end

% Function
function EV = EVT(x, p)
EV = sum(x(:).*p(:));
end